src/actions/city.actions.js

```diff
@@ -0,0 +1,45 @@
+import { cityConstants, citiesConstants} from '../constants';
+import { citiesService } from '../services';
+
+
+export const cityActions = {
+    deleteCity,
+    getByPlaceId,
+    addCity
+};
+
+
+function deleteCity(city){
+    const deletedCity = citiesService.removeCity(city);
+    return dispatch=>{
+        if(deletedCity){
+            dispatch({type: cityConstants.DELETE, city: deletedCity});
+            return;
+        }
+        dispatch({type: cityConstants.ADD,city:undefined});
+    }
+
+} 
+
+function getByPlaceId(city){
+    const selectedCity = citiesService.getCity(city);
+    return dispatch=>{
+        if(selectedCity){
+            dispatch({type: cityConstants.GETBY_PLACEID, city: selectedCity});
+            return;
+        }
+        dispatch({type: cityConstants.GETBY_PLACEID, city: undefined});
```

Comment:
It doesn't make sense to dispatch the same action as in `if` statement but with `undefined` value. You can basically rewrite to these two actions into one `dispatch({type: cityConstants.GETBY_PLACEID, city: selectedCity});` so when selectedCity is undefined or null you automatically get it in your action object.

Reply:
+